packages/server/graphql/mutations/helpers/maybeCreateOneOnOneTeam.ts

```diff
@@ -0,0 +1,124 @@
+import {getUserByEmail} from '../../../postgres/queries/getUsersByEmails'
+import getRethink from '../../../database/rethinkDriver'
+import {RValue} from '../../../database/stricterR'
+import getKysely from '../../../postgres/getKysely'
+import generateUID from '../../../generateUID'
+import createTeamAndLeader from './createTeamAndLeader'
+import IUser from '../../../postgres/types/IUser'
+import inviteToTeamHelper from './inviteToTeamHelper'
+import {GQLContext} from '../../graphql'
+import publish from '../../../utils/publish'
+import {SubscriptionChannel} from '~/types/constEnums'
+import toTeamMemberId from '~/utils/relay/toTeamMemberId'
+
+type OneOnOneTeam = {
+  email: string
+  orgId: string
+}
+
+const getExistingExactTeams = async (userIds: string[], orgId: string) => {
+  const r = await getRethink()
+  const pg = getKysely()
+  const existingTeamIds = await (
+    r
+      .table('TeamMember')
+      // Select teams where both users exists
+      .getAll(r.args(userIds), {index: 'userId'})
+      .filter({isNotRemoved: true}) // Only consider users who are not removed
+      .group('teamId') as RValue
+  )
+    .count()
+    .ungroup()
+    .filter(r.row('reduction').eq(userIds.length))
+    .without('reduction')
+    // Get all team members for each team
+    .eqJoin('group', r.table('TeamMember'), {index: 'teamId'})
+    .zip()
+    .group('teamId')
+    .pluck('userId')
+    .ungroup()
+    // Select only teams with exact given number of team members
+    .filter((row: RValue) => {
+      return row('reduction').count().eq(userIds.length)
```

Comment:
-1 didn't we check this in l32 already?

Reply:
-1 Now that we store whether or not a team is a 1:1 team, we can use that here as well. We can just get a list of `teamId`s where all `userId`s are present and then just check in PG whether the team is a 1:1 team. This way we don't need to count at all. (These big rethink queries scare me)

Reply:
For now I would ignore those 2 cases. If we see decent 1:1 usage, I would later restrict invites for those in the birthday cake version.
I'm open to discuss this.

Reply:
@Dschoordsch I see nothing bad would happen if I ignore these cases for now, so I updated the PR.